File: backend/seeds/populate_user_data.py
import json
import logging
from pathlib import Path

from sqlalchemy import select

from backend.src.middleware.database import SessionLocal
from backend.src.models.player import Player

logger = logging.getLogger(__name__)

BACKEND_DIR = Path(__file__).resolve().parent.parent
PROJECT_ROOT = BACKEND_DIR.parent
DATA_PATH = PROJECT_ROOT / "data" / "players_Data_production.json"

NA_STR = "N/A"


def get_value(obj, key, default=NA_STR):
    val = obj.get(key)
    return default if val is None or val == "" else str(val)


def run_seeds():
    try:
        db = SessionLocal()
        try:
            if db.scalar(select(Player).limit(1)):
                print("Seeding skipped (table already populated).")
                return

            if not DATA_PATH.exists():
                logger.warning("Seed data not found: %s", DATA_PATH)
                return

            with open(DATA_PATH, "r", encoding="utf-8") as f:
                data = json.load(f)

            players_data = data.get("players", [])
            if not players_data:
                return

            for p in players_data:
                db.add(
                    Player(
                        name=get_value(p, "name", NA_STR),
                        age=get_value(p, "age", NA_STR),
                        team=get_value(p, "team", NA_STR),
                        position=get_value(p, "position", NA_STR),
                        jersey_number=get_value(p, "jerseyNumber", NA_STR),
                        preferred_foot=get_value(p, "preferredFoot", NA_STR),
                        height=get_value(p, "height", NA_STR),
                        weight=get_value(p, "weight", NA_STR),
                        image_url=get_value(p, "imageUrl", NA_STR),
                    )
                )
            db.commit()
            print(f"Seeding completed. ({len(players_data)} players)")
        finally:
            db.close()
    except Exception as e:
        print(f"Seeding skipped (database not available): {e}")

File: backend/seeds/populate_user_data.js
import { fileURLToPath } from "url";
import { dirname, join } from "path";
import fs from "fs";
import { pool } from "../src/middleware/database.js";

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);
const BACKEND_DIR = join(__dirname, "..");
const PROJECT_ROOT = join(BACKEND_DIR, "..");
const DATA_PATH = join(PROJECT_ROOT, "data", "players_Data_production.json");

const NA_STR = "N/A";

function getValue(obj, key, defaultVal = NA_STR) {
  const val = obj[key];
  return val == null || val === "" ? defaultVal : String(val);
}

export async function runSeeds() {
  try {
    const client = await pool.connect();
    try {
      const countResult = await client.query("SELECT 1 FROM players LIMIT 1");
      if (countResult.rows.length > 0) {
        console.log("Seeding skipped (table already populated).");
        return;
      }

      let dataPath = DATA_PATH;
      try {
        fs.accessSync(dataPath);
      } catch {
        console.log(`Seed data not found: ${DATA_PATH}`);
        return;
      }

      const raw = fs.readFileSync(dataPath, "utf-8");
      const data = JSON.parse(raw);
      const playersData = data.players || [];
      if (playersData.length === 0) {
        return;
      }

      for (const p of playersData) {
        await client.query(
          `INSERT INTO players (name, age, team, position, jersey_number, preferred_foot, height, weight, image_url)
           VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9)`,
          [
            getValue(p, "name", NA_STR),
            getValue(p, "age", NA_STR),
            getValue(p, "team", NA_STR),
            getValue(p, "position", NA_STR),
            getValue(p, "jerseyNumber", NA_STR),
            getValue(p, "preferredFoot", NA_STR),
            getValue(p, "height", NA_STR),
            getValue(p, "weight", NA_STR),
            getValue(p, "imageUrl", NA_STR),
          ]
        );
      }
      console.log(`Seeding completed. (${playersData.length} players)`);
    } finally {
      client.release();
    }
  } catch (e) {
    console.log(`Seeding skipped (database not available): ${e.message}`);
  }
}
